### ISSUE TITLE
Facebook count not working

### ISSUE THREAD

**Issue by jimmycarlinco** [nygardk/react-share#228]
Currently facebook count is not working anymore. Is this temporary or what?

Thanks for your insights!

**Comment by WejusT1k** [comment]
Same problem

**Comment by trustmedias** [comment]
Same problem. Seems to be a global issue. As I can see, all the websites using it shows "0 Shares" or no counter at all.

Maybe a change into the Facebook graph API, used to get sharing count.

**Comment by keeross** [comment]
Yes, Facebook now requires Access Token of any kind (user/page) to access engagement data.

**Comment by martinKindall** [comment]
So the solution would be to ask as property facebook app's credential on share button.

**Comment by trustmedias** [comment]
That's it.

**Comment by annslittleflower** [comment]
guys can u please provide some code how to solve this?

**Comment by solaymanhaider** [comment]
Did anyone find the solution?

**Comment by codevjs** [comment]
yes some problem. Did anyone find the solution?

**Comment by annslittleflower** [comment]
`
class FacebookShareButton extends Component {

    constructor(props) {
        super(props);

        this.getResponse = this.getResponse.bind(this);
        this.renderWidget = this.renderWidget.bind(this);
    }


  componentDidMount() {

    window.fbAsyncInit = function() {
        FB.init({
            appId      : FB_APP_ID,
            xfbml      : true,
            version    : 'v2.5'
        });
    };
    this.init()
  }

  init() {
    let elem = document.getElementById('facebook-jssdk');
    if (!elem) {
        let fbsharebutton = document.getElementById('fbsharebutton')
        let fbscript = document.createElement('script')
        fbscript.src = '//connect.facebook.net/uk_UA/sdk.js'
        fbscript.id = 'facebook-jssdk'
        fbscript.onload = () => {
            this.renderWidget();
        };
        fbsharebutton.parentNode.appendChild(fbscript)
    } else {
        if(window.FB) {
            this.renderWidget();
        }
    }
  }

  getResponse(response) {
    // this.props.setFBShares(223);
    if (response && !response.error) {
        if (response.engagement && response.engagement.share_count) {
            let sumOfStats = response.engagement.share_count;

            if (response.engagement.comment_count) {
                sumOfStats += response.engagement.comment_count;
            }
            if (response.engagement.comment_plugin_count) {
                sumOfStats += response.engagement.comment_plugin_count;
            }
            if (response.engagement.reaction_count) {
                sumOfStats += response.engagement.reaction_count;
            }

            this.props.setFBShares(sumOfStats);
        }
    }
  }

  renderWidget () {
    FB.api(
        `/v3.2/?id=${this.props.shareUrl}&fields=engagement&access_token=${ACCESS_TOKEN}`,
         this.getResponse
    );
  }

    onShare = () => {
        FB.ui({
            method: 'share',
            href: this.props.shareUrl,
        }, function(response){

        });

    }

  render() {

    return (
        <div
            className="social-btn fb-shares-count"
            onClick={this.onShare}        
        >
            <FbIcon className='social-icon fb'/>

            {this.props.countersVisible && <span
                className="social-share-count fb"
            >
                {this.props.FBShares}
            </span>}
        </div>
    )
  }
} 


const mapStateToProps = (state) => {
    return {
        FBShares: state.article.FBShares,
    }
};

const mapDispatchToProps = dispatch => {
    return {
        setFBShares: (shares) => dispatch(setFBShares(shares)),
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(FacebookShareButton);

`

**Comment by annslittleflower** [comment]
guys please somebody paste in this thread code from above jsfiddle, cannot figure out how to style code in github

**Comment by Trepkos01** [comment]
I opened a pull request with a workaround. Details can be found [here.](https://github.com/nygardk/react-share/pull/237)
